packages/babel-types/src/definitions/utils.ts

```diff
@@ -96,11 +96,18 @@ export function assertEach(callback: Validator): Validator {
   function validator(node: t.Node, key: string, val: any) {
     if (!Array.isArray(val)) return;
 
-    for (let i = 0; i < val.length; i++) {
-      const subkey = `${key}[${i}]`;
+    let i = 0;
+    // For performance
+    const subKey: any = {
+      toString() {
+        return `${key}[${i}]`;
+      },
+    };
```

Comment:
Could you explain this change? I would expect allocating an object to be bad, since then it needs to be garbage collected.

Reply:
The performance gap here seems to come from frequently generating strings without using `subKey`, which is only used when an exception is thrown.
I allocated an object instead of concatenating countless unused strings, so it will be faster.

> Would this have the same effect?

`subKey` should be dynamic along with `i`. :)

Reply:
Oh ok. If you want to do this, we should update `key`'s type to be `string | { toString(): string }`.

Reply:
I tried `key: () => string`, which doesn't work for a static `key` for a non-array element. I'll try `string | { toString(): string }`.